breaks with serde/std

```
   Compiling minimq v0.7.0
error[E0277]: the trait bound `deserializer::Error: StdError` is not satisfied
   --> /home/rj/.cargo/registry/src/index.crates.io-6f17d22bba15001f/minimq-0.7.0/src/de/deserializer.rs:55:27
    |
55  | impl serde::de::Error for Error {
    |                           ^^^^^ the trait `StdError` is not implemented for `deserializer::Error`
    |
note: required by a bound in `packets::_::_serde::de::Error`
   --> /home/rj/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.175/src/de/mod.rs:299:1
    |
299 | declare_error_trait!(Error: Sized + StdError);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `Error`
    = note: this error originates in the macro `declare_error_trait` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `ser::Error: StdError` is not satisfied
   --> /home/rj/.cargo/registry/src/index.crates.io-6f17d22bba15001f/minimq-0.7.0/src/ser/mod.rs:49:28
    |
49  | impl serde::ser::Error for Error {
    |                            ^^^^^ the trait `StdError` is not implemented for `ser::Error`
    |
note: required by a bound in `packets::_::_serde::ser::Error`
   --> /home/rj/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.175/src/ser/mod.rs:186:1
    |
186 | declare_error_trait!(Error: Sized + StdError);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `Error`
    = note: this error originates in the macro `declare_error_trait` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0277`.
error: could not compile `minimq` (lib) due to 2 previous errors
```

--- a/src/de/deserializer.rs
+++ b/src/de/deserializer.rs
@@ -52,6 +52,8 @@ pub enum Error {
     InsufficientData,
 }
 
+impl serde::ser::StdError for Error {}
+
 impl serde::de::Error for Error {
     fn custom<T: core::fmt::Display>(_msg: T) -> Self {
         crate::error!("{}", _msg);

--- a/src/ser/mod.rs
+++ b/src/ser/mod.rs
@@ -46,6 +46,8 @@ pub enum Error {
     Custom,
 }
 
+impl serde::ser::StdError for Error {}
+
 impl serde::ser::Error for Error {
     fn custom<T: core::fmt::Display>(_msg: T) -> Self {
         crate::error!("{}", _msg);

